src/sampler.jl

```diff
@@ -1,43 +1,56 @@
-function logα(H::Real, H_new::Real)
-    return min(0, H - H_new)
+function mh_accept(rng::AbstractRNG, H::AbstractFloat, H_new::AbstractFloat)
+    logα = min(0, H - H_new)
+    return log(rand(rng)) < logα, exp(logα)
 end
+mh_accept(H::AbstractFloat, H_new::AbstractFloat) = mh_accept(GLOBAL_RNG, logα)
 
-function is_accept(logα::Real)
-    return log(rand()) < logα
+function sample(h::Hamiltonian, prop::AbstractProposal, θ::AbstractVector{T}, n_samples::Int; verbose::Bool=true) where {T<:Real}
+    θs = Vector{Vector{T}}(undef, n_samples)
+    Hs = Vector{T}(undef, n_samples)
+    αs = Vector{T}(undef, n_samples)
+    time = @elapsed for i = 1:n_samples
+        θs[i], Hs[i], αs[i] = step(h, prop, i == 1 ? θ : θs[i-1])
+    end
+    verbose && @info "Finished sampling with $time (s)" typeof(h) typeof(prop) EBFMI(Hs) mean(αs)
+    return θs
 end
 
-function sample(h::Hamiltonian, p::AbstractProposal, θ::AbstractVector{T}, n_samples::Int) where {T<:Real}
+function sample(h::Hamiltonian, prop::AbstractProposal, θ::AbstractVector{T}, n_samples::Int, adapter::Adaptation.AbstractAdapter,
```

Comment:
Currently I have two step functions. One for with adapt and one for without.

Reply:
I'm OK to merge them as long as it won't cause performance drop for non-adapt calls